platform/ios/docs/guides/Data-Driven Styling.md

```diff
@@ -0,0 +1,127 @@
+
+#####Exponential:
+
+`MGLInterpolationModelExponential` creates a linear effect based on the values. The key value is the base for interpolation, and the style value is based on where an attribute value falls between two keys.
+
+The stops dictionary below, for example, shows colors that continuously shift from yellow -> orange -> red -> blue -> white based on the attribute value.
+
+```
+  if let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson") {
+            let symbolSource = MGLSource(identifier: "source")
+            let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+
+            let source = MGLShapeSource(identifier: "earthquakes", url: url, options: nil)
+            style.addSource(source)
+
+            let stops = [0 : MGLStyleValue(rawValue: UIColor.yellow),
+                         2.5 : MGLStyleValue(rawValue: UIColor.orange),
+                         5: MGLStyleValue(rawValue: UIColor.red),
+                         7.5 : MGLStyleValue(rawValue: UIColor.blue),
+                         10 : MGLStyleValue(rawValue: UIColor.white)]
+
+            let layer = MGLCircleStyleLayer(identifier: "circles", source: source)
+            layer.circleColor = MGLStyleValue<UIColor>(interpolationMode: .exponential,
+                                                          sourceStops: stops,
+                                                          attributeName: "mag",
+                                                          options: [.defaultValue: MGLStyleValue<UIColor>(rawValue: .green)])
+            layer.circleRadius = MGLStyleValue(rawValue: 10)
+            style.insertLayer(layer, below: symbolLayer)
+        }
+```
+![exponential mode](img/data-driven-styling/exponential.png)
+
+#####Interval
+
+`MGLInterpolationModeInterval` creates a range using the keys stops dictionary. The range is from the given key to just less than the next key. The attribute values that fall into that range are then styled using the style value assigned to that key.
+
+When we use the stops dictionary given above with an interval interpolation mode, we create ranges where earthquakes with a magnitude of 0 to just less than 2.5 be yellow, 2.5 to just less than 5 would be orange, and so on.
+
+```
+layer.circleColor = MGLStyleValue<UIColor>(interpolationMode: .interval,
+                                                          sourceStops: stops,
+                                                          attributeName: "mag",
+                                                          options: [.defaultValue: MGLStyleValue<UIColor>(rawValue: .green)])
+```
+<img width="320" alt="screenshot 2017-02-28 19 14 40" src="https://cloud.githubusercontent.com/assets/12474734/23444472/30d4e8b4-fdea-11e6-8184-ad9631529c74.png">
```

Comment:
Unfortunately we can’t rely on GitHub to host these assets — be sure to commit and use the files required for this guide.